lib/config/migrations/custom/host-rules-migration.ts

```diff
@@ -56,6 +60,27 @@ export class HostRulesMigration extends AbstractMigration {
   }
 }
 
+function validateHostRule(rule: LegacyHostRule & HostRule): void {
+  const { matchHost } = rule;
+  const { hostName, domainName, baseUrl, endpoint, host } = rule;
+  const hostValues = [
+    matchHost,
+    hostName,
+    domainName,
+    baseUrl,
+    endpoint,
+    host,
+  ].filter(is.string);
+  if (hostValues.length > 1) {
```

Comment:
What about checking if their _values_ are duplicated? e.g if someone sets matchHost=abc.com and domainName=abc.com, we could migrate that confidently without throwing an error

Reply:
Should we warn if we found duplicates?

Reply:
Sure, warn if we can keep going, error if we cannot

Reply:
So:
- migrate the values
- check if the values are still unique